servo/connectors/kubernetes.py

```diff
@@ -1438,6 +1438,16 @@ async def delete(self, options:kubernetes_asyncio.client.V1DeleteOptions = None)
                 body=options,
             )
 
+    async def scale_to_zero(self) -> None:
+        """this is used instead of 'delete' as the 'destroy' operation to handle 'on_failure: destroy'.
+        Since the Deployment object is used as a wrapper around an existing k8s object that we did not create,
+        it shouldn't be destroyed. Instead, the deployments pods are destroyed by scaling it to 0 replicas.
```

Comment:
This needs to be part of a seperate PR with a title/branch name associating it with the relevant issue ([ENG-257](https://linear.app/opsani/issue/ENG-257/servoxkubernetes-modify-on-failure-destroy-behavior-for-deployments))

Reply:
It was intended to be "incomplete". We do not want to add this functionality directly into ServoX. FYI: Blake had strong objection to either of the ENG-165 or ENG-166 being integrated into ServoX (and I fully agreed with him on this, too) - they're legacy stuff that is best not to become permanent. On the other hand, the changes here are generic enough and beneficial beyond the specific need for the legacy support - in particular, they allow more to be done in 'before' handlers, which can prove useful in more ways than to simply allow for the functionality described in ENG-165/166 to be implemented.